### ISSUE TITLE
Strange test .only() behavior without --only

### ISSUE THREAD

**Issue by utftu** [oven-sh/bun#3550]
### What version of Bun is running?

0.6.13

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

Create three test files:

`src/number.test.ts`
```ts
import { it } from "bun:test";

it.only("number", () => {});
```

`src/object.test.ts`
```ts
import { it } from "bun:test";

it('object', () => {})
```

`src/string.test.ts`
```ts
import { it } from "bun:test";

it('string', () => {})
```

and run`bun test` (without --only)

### What is the expected behavior?

bun run all test

### What do you see instead?

bun run tests in `src/string.test.ts` and `src/number.test.ts:` but not in `src/object.test.ts`

### Additional information

If you do not place the files in the `src` directory, only `number.test.ts` will be executed.

**Comment by Electroid** [comment]
Thanks for reporting, we'll investigate and fix this. What is likely happening is as soon as the first `.only` is being encountered, the test runner is closing early.

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.1.26
